### DISCUSSION TITLE
Renaming a database table

### DISCUSSION THREAD

**Discussion by amos-kibet** [3Stones-io/omedis#248] [Q&A] [answered]
I did a little research, but I am still unsure what the impact is on renaming an existing database table this way is:
```elixir
rename table(:log_categories), to: table(:activities)
```
Ofcos, I have to recreate/rename indices/constraints/foreign keys to reflect the new change.
My understanding is that existing data is unaffected. Other sources say I should create the new table, copy the data from the old table to the new one, then drop the old table afterwards. What would be a more sane way to do this?

**Comment by wintermeyer** [comment, marked as answer]
IMO the right way is:

PR 1

- Create a new resource (since we want to change the name this is easy)
- Copy all the data from the old to the new resource

PR 2

- Change the relations from other resources to the new one
- Update/Change everything else

PR 3

- Delete the old resource

**Comment by wintermeyer** [comment]
As always: There is the knowledge of the https://elixirforum.com

Ask there how to safely handle this and add a link to that posting here.
